client/landing/gutenboarding/components/locale-context/index.tsx

```diff
@@ -51,11 +51,14 @@ export const ChangeLocaleContextConsumer = ChangeLocaleContext.Consumer;
 export const LocaleContext: React.FunctionComponent = ( { children } ) => {
 	const [ contextLocaleData, setContextLocaleData ] = React.useState< LocaleData | undefined >();
 	const [ localeDataLoaded, setLocaleDataLoaded ] = React.useState( false );
+	const localeSlug = contextLocaleData?.[ '' ]?.localeSlug ?? DEFAULT_LOCALE_SLUG;
+	const wpI18n = React.useMemo( () => createI18n( contextLocaleData ), [ localeSlug ] );
```

Comment:
Similar comment too from my side — `localeSlug`  is listed as a dependency of `useMemo`, but it should probably be `contextLocaleData` ?

Reply:
@p-jackson @ciampo , see why I think it needs to be memoized by the localeSlug here - https://github.com/Automattic/wp-calypso/pull/51406#discussion_r604757485.

Reply:
Got it! Would you mind adding a comment in code to explain the reason of the memoization and why `localeSlug` is the dependency instead of `contextLocaleData` ?
Without any context, opening that file and seeing the ESlint warning may prompt other folks to make changes to it

Reply:
> even though the name of the setLocaleData method suggests that it will set the locale data, what it actually does it to merge the passed locale data with the existing one

Whoa! Good to know, thanks